src/decode.jl

```diff
@@ -21,36 +22,49 @@ julia> path = "test/data/fire.gif"
 julia> img = gif_decode(path)
 60×30×33 Array{RGB{N0f8},3} with eltype RGB{N0f8}
 """
-function gif_decode(filepath::AbstractString)
-    Error = Cint(0)
-    gif = LibGif.DGifOpenFileName(filepath, Ref(Error))
+function gif_decode(filepath::AbstractString; info = false)
+    error = Cint(0)
+    gif = LibGif.DGifOpenFileName(filepath, Ref(error))
 
     try
         gif == C_NULL && error("failed to open the gif file: null pointer")
-        slurpReturn = LibGif.DGifSlurp(gif)
+        slurp_return = LibGif.DGifSlurp(gif)
 
-        if (slurpReturn != LibGif.GIF_OK)
-            println("Failed to read .gif file")
-            return false
+        if (slurp_return == LibGif.GIF_ERROR)
+            error("failed to read .gif file")
         end
-        N = unsafe_load(gif)
-        components = unsafe_wrap(Array, N.SavedImages, N.ImageCount)
-        d = unsafe_load(N.SColorMap)
-        palette = unsafe_wrap(Array, d.Colors, d.ColorCount)
-        final = zeros(RGB{N0f8}, N.SHeight, N.SWidth, N.ImageCount)
-        for i = 1:N.ImageCount
-            #erro handling for this
-            img = unsafe_wrap(Array, components[i].RasterBits, N.SWidth * N.SHeight)
+
+        loaded_gif = unsafe_load(gif)
+
+        img_count = loaded_gif.ImageCount
+        components = unsafe_wrap(Array, loaded_gif.SavedImages, loaded_gif.ImageCount)
+
+        # Gif's are palette based and can have upto 256 colors in a image
+        colormap = unsafe_load(loaded_gif.SColorMap)
+        palette = unsafe_wrap(Array, colormap.Colors, colormap.ColorCount)
+
+        final = zeros(RGB{N0f8}, loaded_gif.SHeight, loaded_gif.SWidth, loaded_gif.ImageCount)
+
+        # read the images
+        for i = 1:img_count
+            img = unsafe_wrap(Array, components[i].RasterBits, loaded_gif.SWidth * loaded_gif.SHeight)
+            desc = components[i].ImageDesc
+            if info == true
+                print("Image $i at [$(desc.Left), $(desc.Top)] and size [$(desc.Width), $(desc.Height)]")
+                (desc.ColorMap !=C_NULL) ? print(" and has a local colormap.\n") : print(" and doesn't have a local colormap.\n")
+            end
```

Comment:
Do you happen to know the `@debug` usage? For instance,

```
JULIA_DEBUG=GIFImages julia --project
```

will print all `@debug` message in GIFImages to the stderr.

See also JuliaLogging: https://julialogging.github.io/how-to/enable-debug/

Reply:
This is quite useful for debugging purposes, using it now 👍🏻